src/test/groovy/org/prebid/server/functional/tests/TargetingSpec.groovy

```diff
@@ -1121,6 +1126,204 @@ class TargetingSpec extends BaseSpec {
         assert targeting["hb_env"] == HB_ENV_AMP
     }
 
+    def "PBS should throw error when price granularity from original request is empty"() {
+        given: "Default bidRequest with empty price granularity"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.targeting = new Targeting(priceGranularity: PriceGranularity.getDefault(UNKNOWN))
+        }
+
+        and: "Account in the DB"
+        def accountAuctionConfig = new AccountAuctionConfig(priceGranularity: PBSUtils.getRandomEnum(PriceGranularityType))
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Request should fail with an error"
+        def exception = thrown(PrebidServerException)
+        assert exception.statusCode == BAD_REQUEST.code()
+        assert exception.responseBody == 'Invalid request format: Price granularity error: empty granularity definition supplied'
+    }
+
+    def "PBS should prioritize price granularity from original request over account config"() {
```

Comment:
`PBS auction should prioritize price granularity from original request over account config`